### ISSUE TITLE
Replacing tensor values on specific indices 

### ISSUE THREAD

**Issue by mongoose54** [keras-team/keras#9407]
I am building my own objective function:

> def my_loss(y_true, y_pred):
>     y_true = tf.reshape(y_true,[-1,256*256,14])     #shape: (num_images, pixels, channels)
>     sum = K.sum(y_true,axis=1)
>     indices = K.tf.where(K.equal(sum, 0))   #Get all the indices where sum of y_true is 0 for each image/channel
>     y_true[ind] = y_pred[ind]     
>     return keras.losses.binary_crossentropy(y_true, y_pred)

What I am not able to figure out how to assign those `ind` of y_true with the values of the same `ind` of y_pred. Basically I cannot do this:  `y_true[ind] = y_pred[ind] `

Any help would be very much appreciated.

**Comment by mongoose54** [comment]
I also tried this simpler version :

```
def my_loss(y_true, y_pred):
    sum = K.sum(y_true,axis=1) # No need to reshape because shape: (num_images, pixels, channels). So do the summation on axis 1 where the pixels are
    y_true = K.switch(K.equal(sum, 0), y_pred, y_true) #Check if sum is 0 replace these elements with y_pred
    return keras.losses.binary_crossentropy(y_true, y_pred)
```

The model compiles but it gives me an error when running the experiment:

`InvalidArgumentError (see above for traceback): Inputs to operation loss/activation_19_loss/Select_1 of type Select must have the same size and shape.  Input 0: [4,917504,14] != input 1: [4,65536,14]`

I don't get where the `917504` came from. I think it came from `K.equal(sum, 0)`
